environment/src/main/java/com/sequenceiq/environment/network/NetworkService.java

```diff
@@ -60,12 +68,37 @@ public boolean hasExistingNetwork(BaseNetwork baseNetwork, CloudPlatform cloudPl
         return environmentNetworkConverterMap.get(cloudPlatform).hasExistingNetwork(baseNetwork);
     }
 
-    @Transactional
-    public BaseNetwork decorateNetworkWithSubnetMeta(Long id, Map<String, CloudSubnet> subnetMetas) {
-        BaseNetwork network = (BaseNetwork) networkRepository.getOne(id);
-        network.setSubnetMetas(subnetMetas);
-        networkRepository.save(network);
-        return network;
+    public Map<String, CloudSubnet> retrieveSubnetMetadata(Environment environment, NetworkDto network) {
+        if (network == null || network.getSubnetIds().isEmpty()) {
+            return Map.of();
+        } else if (AWS.equalsIgnoreCase(environment.getCloudPlatform())) {
+            String regionName = environment.getRegionSet().iterator().next().getName();
+            PlatformResourceRequest prr = new PlatformResourceRequest();
+            prr.setCredential(environment.getCredential());
+            prr.setCloudPlatform(environment.getCloudPlatform());
+            prr.setRegion(regionName);
+            getAwsVpcId(network)
+                    .ifPresent(vpcId -> prr.setFilters(Map.of("vpcId", vpcId)));
+
+            CloudNetworks cnsr = platformParameterService.getCloudNetworks(prr);
+            Set<CloudNetwork> cns = cnsr.getCloudNetworkResponses().get(regionName);
+            return cns.stream()
```

Comment:
It's not easy to keep in mind what is `cns`, `cnsr`, `prr`. Pleaase expand the name of these variables so it becomes human-readable.